File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>




int count_letters();
int count_words();
int count_sentences();

int main(void)
{

    string text = get_string("Text: ");

//turn counts into floats or output is still int
    float letterCount = count_letters(text);
    float wordCount = count_words(text);
    float sentenceCount = count_sentences(text);


    float L = 100 * (letterCount / wordCount);
    float S = 100 * (sentenceCount / wordCount);

//colmeman-liau
    float calc = round(0.0588 * L - 0.296 * S - 15.8);
    int index = round(calc);


    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }



}

int count_letters(string text)
{
    int letters;
    int n = 0;
    letters = strlen(text);

    for (int i = 0; i < letters; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            n++;
        }

    }

    return n;

}

int count_words(string text)
{
    int letters;
    int words = 0;
    letters = strlen(text);

    for (int i = 0; i < letters; i++)
    {
        if ((text[i]) == ' ')
        {
            words++;
        }

    }
    words++;


    return words;
}

int count_sentences(string text)
{
    int letters;
    int sentences = 0;
    letters = strlen(text);

    for (int i = 0; i < letters; i++)
    {
        if (text[i] == '?' || text[i] == '!' || text[i] == '.')
        {
            sentences++;
        }

    }


    return sentences;
}

File: week6/sentimental-readability/readability.py
letters = 0
words = 1
sentences = 0

# input text
text = input("Text: ")

text_length = len(text)

# letters in the text
for i in range(text_length):
    if (text[i].isalpha()):
        letters += 1

# words in text
    if (text[i].isspace()):
        words += 1

# count the number of sentence
    if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
        sentences += 1

calculation = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8
# to calculate Index

index = round(calculation)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")